src/main/java/org/junitpioneer/jupiter/New.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+// TODO: Add @Dir, a shortcut for @New(TemporaryDirectory.class)
+
+// TODO: Add @New(TemporaryFile.class)
+
+// TODO: Add @File, a shortcut for @New(TemporaryFile.class)
+
+@Retention(RetentionPolicy.RUNTIME)
+// TODO: Consider adding and testing ElementType.FIELD
+@Target(ElementType.PARAMETER)
+public @interface New {
+
+	Class<? extends ResourceFactory<?>> value();
+
+	String[] arguments() default {};
```

Comment:
Can be an `Object` array, right? (So we can pass numbers as well.)

Reply:
Never mind. 🙈 (`Object[]` is an invalid type for an annotation member.)

Reply:
> Never mind. :see_no_evil: (Object[] is an invalid type for an annotation member.)

Indeed. :)

It's a right shame, but I can't think of a better way.